jmx-scraper/src/main/java/io/opentelemetry/contrib/jmxscraper/config/JmxScraperConfig.java

```diff
@@ -116,133 +114,50 @@ public boolean isRegistrySsl() {
-    if (propVal.equals(defaultValue)) {
-      properties.setProperty(key, defaultValue);
+  public static JmxScraperConfig fromConfig(ConfigProperties config) {
+    JmxScraperConfig scraperConfig = new JmxScraperConfig();
+
+    Duration exportInterval = config.getDuration(METRIC_EXPORT_INTERVAL);
+    if (exportInterval == null || exportInterval.isNegative() || exportInterval.isZero()) {
+      // if not explicitly set, use default value of 1 minute as defined in specification
+      scraperConfig.samplingInterval = Duration.ofMinutes(1);
+    } else {
+      scraperConfig.samplingInterval = exportInterval;
     }
-    return propVal;
-  }
 
-  private static int getAndSetPropertyIfUndefined(
-      Properties properties, String key, int defaultValue) throws ConfigurationException {
-    int propVal = getProperty(properties, key, defaultValue);
-    if (propVal == defaultValue) {
-      properties.setProperty(key, String.valueOf(defaultValue));
+    String serviceUrl = config.getString(JMX_SERVICE_URL);
+    if (serviceUrl == null) {
+      throw new ConfigurationException("missing mandatory " + JMX_SERVICE_URL);
     }
-    return propVal;
-  }
+    scraperConfig.serviceUrl = serviceUrl;
 
-  /** Will determine if parsed config is complete, setting any applicable values and defaults. */
-  private static void validateConfig(JmxScraperConfig config) throws ConfigurationException {
-    if (isBlank(config.serviceUrl)) {
-      throw new ConfigurationException(SERVICE_URL + " must be specified.");
-    }
-
-    if (isBlank(config.customJmxScrapingConfigPath) && config.targetSystems.isEmpty()) {
+    // TODO: we could support multiple values
+    String customConfig = config.getString(JMX_CUSTOM_CONFIG, "");
+    List<String> targetSystem = config.getList(JMX_TARGET_SYSTEM);
+    if (targetSystem.isEmpty() && customConfig.isEmpty()) {
       throw new ConfigurationException(
-          CUSTOM_JMX_SCRAPING_CONFIG + " or " + TARGET_SYSTEM + " must be specified.");
-    }
-
-    if (!config.targetSystems.isEmpty()
-        && !AVAILABLE_TARGET_SYSTEMS.containsAll(config.targetSystems)) {
-      throw new ConfigurationException(
-          String.format(
-              "%s must specify targets from %s", config.targetSystems, AVAILABLE_TARGET_SYSTEMS));
-    }
-
-    if (OTLP_METRICS_EXPORTER.equalsIgnoreCase(config.metricsExporterType)
-        && isBlank(config.otlpExporterEndpoint)) {
-      throw new ConfigurationException(OTLP_ENDPOINT + " must be specified for otlp format.");
-    }
-
-    if (config.intervalMilliseconds < 0) {
-      throw new ConfigurationException(INTERVAL_MILLISECONDS + " must be positive.");
+          "at least one of '" + JMX_TARGET_SYSTEM + "' or '" + JMX_CUSTOM_CONFIG + "' must be set");
     }
+    targetSystem.forEach(
+        s -> {
+          if (!AVAILABLE_TARGET_SYSTEMS.contains(s)) {
+            throw new ConfigurationException("unsupported target system: '" + s + "'");
```

Comment:
Doesn't this get hidden because it's buried within a closure within `forEach`?

Could consider switching to `if(!targetSystem.stream().allMatch(AVAILABLE_TARGET_SYSTEMS.contains(s)) ...` or similar? Maybe a filter if the detailed logging of which system is necessary.

Reply:
The exception does not get hidden here, this case is covered by the `JmxScraperConfigTest#shouldFailValidation_invalidTargetSystem` test case.